### ISSUE TITLE
Cannot discover other devices

### ISSUE THREAD

**Issue by Tienisto** [localsend/localsend#527]
Because computer networks are complex, it is not always possible to have a functioning discovery that works offline.
There are just too many issues opened by users that is hard to follow.

Please only create a new issue if you know the exact details why this error happens. Only "it does not work in my wifi" is NOT enough. (You can comment here if you wish).

General solutions:
- it may not work in your wifi because AP isolation is enabled which prohibits communication and discovery between wifi participants
- different configuration (port, encryption)

**Comment by archerallstars** [comment]
I share from my Android phone's hotspot to Linux without any issue. But I want to add that, **if the phone is connecting to VPN, sharing via hotspot won't work.**

It took me a while to figure that 😂

**Comment by Joackk** [comment]
I had a problem recently. localsend on the physical machine cannot detect localsend on the virtual machine. First I started localsend on my desktop machine, then I started the virtual machine on the desktop and started localsend in the virtual machine. But they can't detect each other, I'm using bridge mode, ping can communicate. And the firewall has been cleared. To test the problem, I turned localsend on again on my laptop, and both he and the desktop could find each other, but they still couldn't find localsend on the virtual machine. I then enabled a virtual machine in my notebook as well and turned on localsend. VMS on two different physical machines can discover each other. But they were unable to discover the physical machine. They're all in bridge mode, they're all on the same LAN, and ping works fine. Why is that?

**Comment by Joackk** [comment]
One more note: Both virtual machines and physical machines can detect localsend on the phone normally

**Comment by PerMildner** [comment]
Summary: Macs cannot see each other but both are visible to Android. Using a portable install works fine. Possibly caused by setting files copied from old mac to new mac during account migration.

I have LocalSend 1.16.1 on two MacBooks (14.7.1 AS, 14.7.1 Intel). Same WIFI.

They cannot see each other. However, an Android phone (LocalSend 1.15.4) on the same WIFI can see them both. Also, the macs can see the Android phone.

I copied LS to a new folder on the AS Mac, and created an empty `settings.json` file to get a "portable" install. If I start the portable LS  on the AS Mac the AS Mac becomes visible on the Intel Mac, and the Intel Mac can send files to the AS Mac.

Possibly relevant:
The following is all about the versions of LS installed in /Applications/, i.e. not about the "portable" install.

The AS Mac started out with the files from the older Intel Mac (Apple's account migration or whatever). Presumably this made the older LocalSend settings end up also on the AS Mac.

When I started LS on the two machines, they both used the _same_ "Blueberry Banjo" (or some such) name, which I guess was because of some common state in the copied settings. Changing the name in the LS settings on the Intel Mac did not help (I restarted both LS servers etc).

I wanted to try to delete the saved LS state on the AS Mac, but the files I found were some sandbox directory that even `sudo` is not allowed to delete or rename.

**Comment by Tienisto** [comment]
Maybe it helps if you toggle "Local Network" permission in the system settings for LS.

How did you got the portable version? The latest DMG should be sandboxed also

**Comment by dwelden** [comment]
I have the same issue. iPhone can see and send files to Linux. Linux cannot see the iPhone. Entering the iPhone IP address manually gives error:

`[RhttpConnectionException] Connection error. URL: https://192.168.12.56:53317/api/localsend/v1/info (hyper_util::client::legacy::Error(Connect, ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" })))`

Linux: 1.16.1 (56) Flatpak
iPhone: 1.16.2 (57)
iOS: 17.7.2
iOS setting Privacy & Security/Local Network is toggled On for LocalSend

**Comment by tukykarmakar** [comment]
I have a very unusual setup with **one desktop PC** and **two Android phones**--
- The PC is dual-booting Windows 10 and [Nobara Linux](https://nobaraproject.org/)
- I have a wifi router which I don't want to use when using my PC, so I got a wifi adapter.
- The wifi adapter is **TP Link Archer T6E AC1300**, which uses **Broadcom BCM4360 (rev 03)** chipset, according to the output of `lspci -vnn -d 14e4:` run on Linux.
- When using the PC, I connect my ethernet cable directly to my PC and use the wifi adapter to create hotspot for using the phones when needed.
- But **hotspot doesn't work on Linux** because Broadcom doesn't support it, only **connecting to wifi networks is supported** by the card.
- I have Localsend installed on both Windows 10 and Linux. I'm using Localsend Flatpak on Linux.

On Windows, I can turn on the hotspot and connect my Android phones and use Localsend without any issues. This is is my desired use-case. As hotspot isn't supported on Linux, I have to create hotspot on the phones and then connect my PC to them to use Localsend.

The approach above used to work well until recently, where **Localsend isn't able to detect the phones when my ethernet connection is active**. However, my phone is able to see my PC, my PC can receive without issues, but **I need to use the phone's IP address directly to send from PC** because it can't detect my phone. Whenever I disconnect or deactivate my ethernet connection, Localsend is able to find my phone again.

I saw an issue regarding discovery in networks with large subnets, but I don't know why it's happening now when it never happened before.

https://github.com/user-attachments/assets/9481e00f-5d21-4501-aae8-500b0664a593

**Comment by Mapaler** [comment]
#2723  is a rare situation, but there is a known solution.

**Comment by JanUwU42** [comment]
I had the Issue that CachyOS and iPhone didnt find each other, the solution was to allow LocalSend in my Firewall
`sudo ufw allow 53317/tcp`

`sudo ufw reload`
Maybe this will help someone having the same problem :D

**Comment by sasha-computer** [comment]
@JanUwU42 that fixed it for me on cachyOS as well, thank you!

**Comment by DantesNoWar** [comment]
Hi, LocalSend don't work on all my devices: Windows 10 PC (2013), Win10 PC (2020/21), and Xiaomi HyperOS 2 (Android 15).
I have a single Wifi router (ZTE X388X) connection with no AP isolation active AFAIK. I has a single channel both for 2.4 and 5Ghz.
I have set Quad9 DNS at a router level for all devices. I've tried with "Automatic Resolution" on both the Win10 PCs and tried to tweak with LocalSend settings on all devices, but nothing has changed.
What can I do? :(

**Comment by pyeung99** [comment]
I use an Android phone. My wife uses an iPhone.  We have installed the app LocalSend on both phones. The two phones are on the same home network.

I can transfer picture files from the iPhone to the Android phone with LocalSend. I am unable to transfer any files from my Android phone to my wife’s iPhone. The name corresponding to the iPhone does not appear under the Nearby devices section on the Send screen of the LocalSend app on the Android phone; therefore, no send request can be initiated.

The "AP Isolation" setting on my home router is disabled.

How can I transfer files from my Android phone to my wife’s iPhone using LocalSend?

**Comment by andradei** [comment]
@JanUwU42 That worked for me on CachyOS as well!
What's weird is LocalSend on my mobile device found it the very first time. Then it stopped finding my linux computer until I allowed it with `ufw`.

**Comment by DantesNoWar** [comment]
I'm late but I solved my issue. I was using a VPN wth killswitch on one device and forget about it.

**Comment by KevsterAmp** [comment]
> I use an Android phone. My wife uses an iPhone.  We have installed the app LocalSend on both phones. The two phones are on the same home network.
> 
> I can transfer picture files from the iPhone to the Android phone with LocalSend. I am unable to transfer any files from my Android phone to my wife’s iPhone. The name corresponding to the iPhone does not appear under the Nearby devices section on the Send screen of the LocalSend app on the Android phone; therefore, no send request can be initiated.
> 
> The "AP Isolation" setting on my home router is disabled.
> 
> How can I transfer files from my Android phone to my wife’s iPhone using LocalSend? 

@pyeung99 i have experienced a similar issue. I have solved it by opening a hotspot on android and connecting the iphone.

Iphone now becomes visible on localsend and I have transferred files from android to iphone succesfully

**Comment by octopusgt** [comment]
I can confirm a reproducible discovery regression on a Google Pixel 5 with LocalSend 1.18.1 (61).

Setup:

* Google Pixel 5: LocalSend 1.18.1 (61)
* Windows PCs on the same Wi-Fi/LAN:

  * LocalSend 1.16.1 (56)
  * LocalSend 1.16.1 (56)
  * LocalSend 1.17.0 (58)
* Same subnet (192.168.1.x), Windows networks set to Private.
* Default LocalSend port 53317.

Symptoms with LocalSend 1.18.1 on Android:

* The Pixel may appear briefly in Nearby Devices, but pressing Refresh makes it disappear and it often cannot be discovered again.
* Direct communication by IP works, so basic TCP connectivity is OK.
* Discovery Debugging on the Pixel can remain completely empty when a Windows client performs a new discovery/refresh.
* Rebooting the phone, restarting LocalSend, clearing app storage, reinstalling LocalSend, forgetting/rejoining Wi-Fi, and limiting LocalSend to wlan0 did not fix it.
* A second Pixel 5 running LocalSend 1.18.1 has also shown the same problematic behavior.

Most importantly: **downgrading the affected Pixel 5 from LocalSend 1.18.1 to 1.17.0 immediately restores normal device discovery.** The phone becomes discoverable by the Windows clients again, with no router or network changes.

This appears to be a regression introduced in the 1.18.x Android networking/discovery implementation.

I can provide Discovery Debugging screenshots/logs and additional tests if useful.

Hope this helps! Thanks.

P.S. LocalSend rocks! Fantastic app.
